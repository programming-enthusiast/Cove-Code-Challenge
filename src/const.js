export const API_URL = "https://cove-coding-challenge-api.herokuapp.com/reservations"