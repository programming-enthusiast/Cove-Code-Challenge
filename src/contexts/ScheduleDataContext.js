import React from 'react'

export default React.createContext({
  scheduleData: null,
  setSchedulData: undefined,
  roomList: [],
  setRoomList: undefined,
  currentRoomId: null,
  setCurrentRoomId: undefined
})