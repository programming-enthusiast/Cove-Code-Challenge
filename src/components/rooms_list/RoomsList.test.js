import React from "react"
import RoomList from "./RoomsList"
import ScheduleDataContext from "../../contexts/ScheduleDataContext"
import renderer from "react-test-renderer"
import { ROOM_LIST, SCHEDULE_DATA } from '../../const'

describe("RoomList Component", () => {
  let renderedComponent

  let contextValue = { scheduleData: SCHEDULE_DATA, roomList: ROOM_LIST}

  beforeAll(() => {
    renderedComponent = renderer.create(
      <ScheduleDataContext.Provider value={contextValue}>
        <RoomList/>
      </ScheduleDataContext.Provider>
      )
  })

  it("should render correctly", () => {
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  })
})
