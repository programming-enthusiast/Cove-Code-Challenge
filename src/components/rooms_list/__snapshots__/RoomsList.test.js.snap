// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomList Component should render correctly 1`] = `
<div
  className="room-list__container"
>
  <div
    className="room-list__title"
  >
    <h1>
      Select a room to view reservations
    </h1>
  </div>
  <div
    className="room-list__grid_view"
  >
    <div
      className="room-list__listing-card"
      onClick={[Function]}
    >
      <div
        className="room-list__listing-card__img-aspect-ratio-box"
      >
        <img
          className="room-list__listing-card__image"
          src="https://staging.cove.is/parse/files/hRKEvW2lN74k5nCg6p2XtmiWRNHycE2pHpXpELMX/f96efd3f11aadb34135bb1f0aecf9667_Quincy%20Room.jpg"
        />
      </div>
      <div
        className="room-list__listing-card-info-section"
      >
        <div
          className="room-list__listing-card-info-section__room-name"
        >
          Room A
        </div>
      </div>
    </div>
    <div
      className="room-list__listing-card"
      onClick={[Function]}
    >
      <div
        className="room-list__listing-card__img-aspect-ratio-box"
      >
        <img
          className="room-list__listing-card__image"
          src="https://staging.cove.is/parse/files/hRKEvW2lN74k5nCg6p2XtmiWRNHycE2pHpXpELMX/d0d19da4aa88734291279f5fe7a836e7_Wakefield%20Room.jpg"
        />
      </div>
      <div
        className="room-list__listing-card-info-section"
      >
        <div
          className="room-list__listing-card-info-section__room-name"
        >
          Room B
        </div>
      </div>
    </div>
  </div>
  <div
    className="room-list__page-stepper__wrapper"
  >
    <div
      className="room-list__page-stepper__container"
    >
      <div
        className="room-list__page-stepper__previous-btn"
        onClick={[Function]}
      >
        <svg
          fill="rgb(226, 226, 226)"
        >
          prev_btn.svg
        </svg>
      </div>
      <div
        className="room-list__page-stepper__spacer"
      />
      <div
        className="room-list__page-stepper__page selected"
      >
        1
      </div>
      <div
        className="room-list__page-stepper__spacer"
      />
      <div
        className="room-list__page-stepper__next-btn"
        onClick={[Function]}
      >
        <svg
          fill="rgb(226, 226, 226)"
        >
          next_btn.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
